### DISCUSSION TITLE
lint:twig Error in twig-bridge

### DISCUSSION THREAD

**Discussion by matthieuleorat** [symfony/symfony#44264] [Q&A] [answered]
Hello,
on a symfony 5.3.12 project, i tried the command php bin/console lint;twig
I was surprised to see some errors inside the twig-bridge bundle:
```shell
php bin/console lint:twig

  ERROR  in /srv/app/vendor/symfony/twig-bridge/Resources/views/Email/zurb_2/notification/content_markdown.html.twig (line -1)
  >> The "markdown_to_html" filter is part of the MarkdownExtension, which is not installed/enabled; try running "composer require twig/markdown-extra". 
  ERROR  in /srv/app/vendor/symfony/twig-bridge/Resources/views/Email/zurb_2/notification/body.html.twig (line -1)
  >> The "inky_to_html" filter is part of the InkyExtension, which is not installed/enabled; try running "composer require twig/inky-extra". 
  ERROR  in /srv/app/vendor/symfony/twig-bridge/Resources/views/Email/zurb_2/notification/content_markdown.html.twig (line -1)
  >> The "markdown_to_html" filter is part of the MarkdownExtension, which is not installed/enabled; try running "composer require twig/markdown-extra". 
  ERROR  in /srv/app/vendor/symfony/twig-bridge/Resources/views/Email/zurb_2/notification/body.html.twig (line -1)
  >> The "inky_to_html" filter is part of the InkyExtension, which is not installed/enabled; try running "composer require twig/inky-extra".
```

This bridge is required by the following packages:
```shell
composer why symfony/twig-bridge
symfony/debug-bundle      v5.3.4   requires   symfony/twig-bridge (^4.4|^5.0)  
symfony/form              v5.3.11  conflicts  symfony/twig-bridge (<4.4)       
symfony/framework-bundle  v5.3.11  conflicts  symfony/twig-bridge (<4.4)       
symfony/http-kernel       v5.3.12  conflicts  symfony/twig-bridge (<5.0)       
symfony/twig-bundle       v5.3.10  requires   symfony/twig-bridge (^5.3)
```

I an wondering if it has to be fixed somewhere in a symfony package ? 
Should twig-brige package would have twig/markdown-extra and twig/inky-extra as dependencies ?

An issue already exist on [symfony/recipes](https://github.com/symfony/recipes/issues/980).

I would be happy  to contribute on this issue, but i don't know how to fix it.

**Comment by stloyd** [comment, marked as answer]
It's not an issue, you should **not** lint whole project (with vendors), you should **only** lint your templates so it should be:
```bash
php bin/console lint:twig src/ templates/
```

**Reply by matthieuleorat** [reply]
Yes i understand, I should only lint my own templates.
But still, twig-bridge package use the twig filter [markdown_to_html](https://github.com/symfony/twig-bridge/blob/5.3/Resources/views/Email/zurb_2/notification/content_markdown.html.twig), provided by the package [twigphp/markdown-extra](https://github.com/twigphp/markdown-extra/blob/2.x/MarkdownExtension.php)

I think symfony/twig-bridge should have `twig/markdown-extra` and `twig/inky-extra` as long as it use their extensions

**Reply by stof** [reply]
Well, the twig-bridge uses the `|markdown_to_html` filter only for an optional feature, when you want to use the markdown-based template for the notifier component. That's why it is not a required dependency. When you actually try to use that feature, you get a helpful error message telling you which package you need to install to provide the optional dependency.

**Reply by matthieuleorat** [reply]
ok, thank for the explaination :).
